### PR TITLE
Contribution updates

### PR DESCRIPTION
**Pull request by simskij**
Adds a PR template asking the user to discuss changes either on slack or in an issue before getting started with their implementation. Adds the same text to the contribution guidelines as well.

### CODE REVIEW

File: .github/pull_request_template.md
Comment (na--) [inline .github/pull_request_template.md:15]: Maybe add a sentence at the end, something like
> If you've already done all of that, please go ahead and submit your PR and thank you for the contribution!

Doesn't sound quite right, but :man_shrugging:

**Review by na-- — APPROVED** [review]

File: .github/pull_request_template.md
Comment (MStoykov) [inline .github/pull_request_template.md:13]: How about the community forum ?

**Review by MStoykov — APPROVED** [review]

File: .github/pull_request_template.md
Comment (simskij) [reply .github/pull_request_template.md:13]: I can add that if it makes sense for you. 😅  I assumed (maybe incorrectly) that you, as core contributors, would prefer gh issues and slack.

File: .github/pull_request_template.md
Comment (simskij) [reply .github/pull_request_template.md:15]: See proposed change 😄

File: .github/pull_request_template.md
Comment (na--) [reply .github/pull_request_template.md:13]: for most things, github issue > community forum > slack - slack would only win over the community forum if near-real-time communication would be useful, for example to debug a tricky or urgent bug, or something like that

Comment (simskij) [reply .github/pull_request_template.md]: Done. Reapprove and i'll merge it 👍

**Review by na-- — APPROVED** [review]
LGTM, please squash when merging

**Pull request merged**
